data-integration/main.bal

```diff
@@ -0,0 +1,102 @@
+import data_integration.store;
+import ballerina/http;
+import ballerina/persist;
+
+configurable int port = 9090;
+
+service /taskmanager on new http:Listener(port) {
+    private final store:Client dbClient;
+
+    function init() returns error? {
+        self.dbClient = check new();
+    }
+
+    isolated resource function post employee(store:EmployeeInsert employee)
+            returns http:Created|http:Conflict|http:InternalServerError {
+        int[]|persist:Error result = self.dbClient->/employees.post([employee]);
+        if result is persist:Error {
+            if result is persist:AlreadyExistsError {
+                return http:CONFLICT;
+            }
+            return <http:InternalServerError>{body: result.message()};
+        }
+        return http:CREATED;
```

Comment:
Since the non-error case is also just returning, we can also do

```suggestion
        if result is persist:AlreadyExistsError {
            return http:CONFLICT;
        }

        if result is persist:Error {
            return <http:InternalServerError>{body: result.message()};
        }

        return http:CREATED;
```

Reply:
Alternatively, if we can extract out common logic that returns an appropriate response on `persist:Error`, I'm OK with extracting that bit out to a function too.